invenio_notifications/models.py

```diff
@@ -8,36 +8,73 @@
 
 """Models used for notifications."""
 
-from dataclasses import dataclass
+from dataclasses import asdict, dataclass, field
 from datetime import datetime
 
 
 @dataclass
 class Notification(dict):
-    """Notification class."""
+    """Notification base class."""
 
     type: str  # event type e.g comment_edit, new_invitation etc
     data: dict  # depending on the type. dump of a record, community, etc.
-    recipients: list  # list of user dumps
     trigger: dict  # info about who triggered (and dump thereof), if it was manual or automatic
     timestamp: str  # when the action happened
 
-    def __init__(self, type="", data={}, recipients=None, trigger={}, **kwargs):
+    def __init__(self, type="", data=None, trigger=None, **kwargs):
         """Constructor."""
         self.type = self["type"] = type
```

Comment:
we could overwrite `__getitem__` to avoid having duplicate info in an attribute and a key dict?
```python
>>> class Test(dict):
...     def __init__(self):
...             super().__init__()
...             self.test = "123"
...     def __getitem__(self, key):
...             return getattr(self,key)
...
>>> test = Test()
>>> test["test"]
'123'
```

this might affect the dump method, might need to modify `__setitem__` too